x-pack/plugins/apm/server/lib/helpers/get_random_sampler_seed/index.ts

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { KibanaRequest } from '@kbn/core/server';
+import seedrandom from 'seedrandom';
+import { APMRouteHandlerResources } from '../../..';
+
+export async function getRandomSamplerSeed({
+  security,
+  request,
+}: {
+  security: APMRouteHandlerResources['plugins']['security'];
+  request: KibanaRequest;
+}) {
+  if (security) {
```

Comment:
I'd prefer if we use a hardcoded seed if we cannot generate one based on the current user, for whatever reason. That makes it harder to forget about passing it down.

Reply:
Why don't we send `seed` together with the `probability` from the client and make it required? We can store it in local/session storage to keep it stable.

Reply:
Default value is fine.

Reply:
Added in https://github.com/elastic/kibana/pull/133789/commits/49fd905f697cc82971b4d2b7906d4246fa090a27